### PR TITLE
Fix layout when includeViewExtension enabled

### PR DESCRIPTION
**Pull request by jorgevrgs**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -107,12 +109,23 @@ function fastifyView (fastify, opts, next) {
   })
 
   function getPage (page, extension) {
-    if (viewExt) {
-      return `${page}.${viewExt}`
-    } else if (includeViewExtension) {
-      return `${page}.${extension}`
+    const cacheId = `getPage-${page}-${extension}`
+    let result = lru.get(cacheId)
```
Comment (mcollina) [inline index.js:113]: Please use a difference/new cache for this, call it `pageLRU`

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you please add a unit test?

File: index.js (hunk 1 above)
Comment (jorgevrgs) [reply index.js:113]: Done

Comment (jorgevrgs) [reply index.js:113]: Ohhh!!! sorry. It was my mistake, isn't it? Is it about the concatenated name of the variable name?

File: index.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ const readdirSync = require('fs').readdirSync
 const resolve = require('path').resolve
 const join = require('path').join
 const basename = require('path').basename
+const dirname = require('path').dirname
+const extname = require('path').extname
```
Comment (zekth) [inline index.js:13]: why not refactoring using destructuring syntax? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Comment (jorgevrgs) [reply index.js:13]: Yes. I considered it, but I was just trying to preserve the same structure. So, I'll modify it.

> why not refactoring using destructuring syntax? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

**Comment by jorgevrgs** [comment]
> Can you please add a unit test?

I'm on it but I found that it's necessary to extend the solution for `hasAccessToLayoutFile` because tests were approved but taking the `layout.hbs` file instead of the `.ejs` file. So let me check that part.

**Comment by jorgevrgs** [comment]
> > Can you please add a unit test?
> 
> I'm on it but I found that it's necessary to extend the solution for `hasAccessToLayoutFile` because tests were approved but taking the `layout.hbs` file instead of the `.ejs` file. So let me check that part.

Results (in process to upload):

Original: 97.5% 15335
Updated: 97.5% 15327

File: index.js (hunk 1 above)
Comment (mcollina) [reply index.js]: You are using an LRU with mixed content. Instead, the content should be omogeneous. Instantiate a new lru cache and use that one for this purpose.

Comment (jorgevrgs) [reply index.js]: > You are using an LRU with mixed content. Instead, the content should be omogeneous. Instantiate a new lru cache and use that one for this purpose.

Is it possible using the same `maxCache` option?
``` js
  const lru = HLRU(opts.maxCache || 100)
  const pageLRU = HLRU(opts.maxCache || 100)
```

Or what's the recommendation there?

Comment (zekth) [reply index.js]: Like this is ok as pageLRU will not be big as it stores only a tiny string.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by zekth — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**